packages/askui-nodejs/src/execution/inference-client.ts

```diff
@@ -34,7 +34,8 @@ export class InferenceClient {
       ? urljoin(versionedBaseUrl, 'workspaces', workspaceId)
       : versionedBaseUrl;
     this.urls = {
-      inference: urljoin(url, 'inference'),
+      // V4's only end part is different
+      inference: apiVersion === 'v4' ? urljoin(url, 'predict') : urljoin(url, 'inference'),
```

Comment:
why are you not using 'inference' anymore?

Reply:
We decide to move away from the current implementation of model composition and URL to make it simpler and user readable. Can be found in [Ticket](https://askyourui.atlassian.net/browse/ML-465)